### PR TITLE
Create TrunkBasedVersionStrategy implementation

### PR DESCRIPTION
**Pull request by HHobeck**
## Description
Replace the version mode Mainline in 6.x #3601

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: src/GitVersion.Core/Configuration/EffectiveBranchConfiguration.cs [hunk 1]
```diff
@@ -7,4 +8,12 @@ public record EffectiveBranchConfiguration(EffectiveConfiguration Value, IBranch
     public IBranch Branch { get; } = Branch.NotNull();
 
     public EffectiveConfiguration Value { get; } = Value.NotNull();
+
+    public NextVersion CreateNextVersion(BaseVersion baseVersion, SemanticVersion incrementedVersion)
```
Comment (arturcic) [inline src/GitVersion.Core/Configuration/EffectiveBranchConfiguration.cs:12]: This method was already moved to `NextVersionCalculator` where the usage is, please that one instead, I prefer to have this one POCO

**Comment by HHobeck** [comment]
Please go ahead and merge the changes to main. Thank you!

**Comment by arturcic** [comment]
@HHobeck please rebase on top of `main` and then we can start the review.

**Comment by arturcic** [comment]
@HHobeck maybe you can create a branch with your changes and create a new PR out of it, and just make a reference to this PR

**Pull request closed without merging**